src/api/endpoints/ApiEndpoint.ts

```diff
@@ -2,37 +2,39 @@ import axios, { AxiosError, AxiosRequestConfig } from "axios";
 import { Logger } from "ts-log";
 
 import { HeaderProvider } from "../../Func";
+import { DefaultApiConfig } from "../ApiConfig";
 import ApiResponse from "../ApiResponse";
 import TokenManager from "./auth/TokenManager";
 
 export default abstract class ApiEndpoint {
-  public readonly baseUrl: string;
+  public readonly config: DefaultApiConfig;
 
   protected constructor(
-    baseUrl: string | undefined,
+    config: DefaultApiConfig,
     public readonly tokenManager: TokenManager,
     protected readonly log?: Logger,
     private readonly headerProvider?: HeaderProvider,
   ) {
-    this.baseUrl = baseUrl ?? "https://api.medrunner.space";
+    this.config = config;
   }
 
   protected abstract endpoint(): string;
 
   protected endpointUrl(): string {
-    return `${this.baseUrl}/${this.endpoint()}`;
+    return `${this.config.baseUrl}/${this.endpoint()}`;
   }
 
   private async headersForRequest(noAuthentication: boolean): Promise<AxiosRequestConfig> {
     const config: AxiosRequestConfig = {
-      baseURL: this.baseUrl,
+      baseURL: this.config.baseUrl,
       headers: {},
+      withCredentials: this.config.cookieAuth,
     };
 
     if (config.headers !== undefined) {
       if (!noAuthentication) {
         const accessToken = await this.tokenManager.getAccessToken("API makeRequest");
-        if (accessToken !== undefined) {
+        if (accessToken !== undefined || !this.config.cookieAuth) {
```

Comment:
If access token is undefined we don't want any way of ending up in here right? Otherwise the header will just be `Bearer `.

Reply:
Yeah true, I've removed the second check and made sure that we don't have an access token returned if we use cookieAuth, so we shouldn't go into that if statement if we use cookieAuth